muffnn/mlp/base.py

```diff
@@ -301,6 +301,26 @@ def _set_up_graph(self):
         if self._transform_layer_index == -1:
             self._transform_layer = t
 
+        # Prediction gradients (e.g., for analyzing the importance of features)
+        # We use the top layer before the output activation function
+        # (e.g., softmax, sigmoid) following
+        # https://arxiv.org/pdf/1312.6034.pdf
+        if self.is_sparse_:
+            self._prediction_gradient = None
+        else:
+            output_shape = self.output_layer_.get_shape()
+            if len(output_shape) == 1:
+                self._prediction_gradient = tf.gradients(
+                    t, self._input_values)[0]
+            else:
+                # According to the tf.gradients documentation, it looks like
+                # we have to compute gradients separately for each output
+                # dimension and then stack them for multiclass/label data.
+                self._prediction_gradient = tf.stack([
+                    tf.gradients(t[:, i], self._input_values)[0]
+                    for i in range(self.output_layer_.get_shape()[1])
```

Comment:
(nitpick) Since we already defined `output_shape = self.output_layer_.get_shape()` this line might read a little easier if we used that variable.

Reply:
oh, good catch, thanks